### PR TITLE
or-tools: 9.4 -> 9.7

### PR DESCRIPTION
**Pull request by wegank**
## Description of changes

Not to 9.8 as it requires absl-py 2.0 and #264902.

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by wegank** [comment]
@ofborg build or-tools or-tools.passthru.tests

**Comment by wegank** [comment]
@ofborg eval

File: pkgs/development/libraries/science/math/or-tools/default.nix [hunk 1]
```diff
@@ -18,53 +18,57 @@
 , zlib
 }:
 
+let
+  pybind11_protobuf = fetchFromGitHub {
+    owner = "pybind";
+    repo = "pybind11_protobuf";
+    rev = "b713501f1da56d9b76c42f89efd00b97c26c9eac";
```
Comment (andersk) [inline pkgs/development/libraries/science/math/or-tools/default.nix:25]: According to https://github.com/google/or-tools/blob/v9.7/WORKSPACE#L228, this should be `5baa2dc9d93e3b608cde86dfa4b8c63aeab4ac78`.

File: pkgs/top-level/all-packages.nix [hunk 2]
```diff
@@ -39314,12 +39314,10 @@ with pkgs;
   or-tools = callPackage ../development/libraries/science/math/or-tools {
     stdenv = if stdenv.isDarwin then overrideSDK stdenv "11.0" else stdenv;
     python = python3;
-    protobuf = protobuf_21;
-    # or-tools builds with -std=c++20, so abseil-cpp must
-    # also be built that way
-    abseil-cpp = abseil-cpp_202206.override {
+    protobuf = protobuf_23;
+    abseil-cpp = abseil-cpp.override {
```
Comment (andersk) [inline pkgs/top-level/all-packages.nix:39318]: OR-Tools [requires a specific version of abseil-cpp](https://github.com/google/or-tools/blob/v9.7/Dependencies.txt#L2), and historically it almost always fails to build with versions newer than the specified one, so we should still pin this to `abseil-cpp_202301`.

File: pkgs/development/libraries/science/math/or-tools/default.nix (hunk 1 above)
Comment (wegank) [reply pkgs/development/libraries/science/math/or-tools/default.nix:25]: I had to use `b713501f1da56d9b76c42f89efd00b97c26c9eac` for the build to succeed, probably due to some mismatches as in google/or-tools#4027.

**Comment by wegank** [comment]
This should also drop `or-tools` from the list of failures in #277049, I think.

**Comment by wegank** [comment]
Reopening the PR as or-tools is [failing on master](https://hydra.nixos.org/job/nixpkgs/trunk/or-tools.x86_64-linux). I'll do the next bump to 9.9 in other PR, whose diff is way more significant.

**Pull request merged**